test-source/test/ceylon/io/testSockets.ceylon

```diff
@@ -75,8 +79,9 @@ void writeRequestInPipeline(String request, Socket socket) {
     value requestBuffer = ByteBuffer.ofSize(200);
     value encoder = ascii.chunkEncoder();
     value input = CharacterBuffer(request);
+    value iter = input.iterator();
     while(input.hasAvailable || !encoder.done){
-        encoder.convert(requestBuffer, input);
+        encoder.convert(requestBuffer, iter.next);
         // flip and flush the request buffer
         requestBuffer.flip();
         // write it all, blocking
```

Comment:
This usage model is ugly! `input.hasAvailable` is used to see if the `input.iterator()` has elements left, which is a pattern that is dependent on the internals of `CharacterBuffer`. Perhaps this would be better:
```ceylon
    value provider = iter.next;
    while(0 != encoder.convert(requestBuffer, provider)) {
        encoder.convert(requestBuffer, iter.next);
        // flip and flush the request buffer
        requestBuffer.flip();
        // write it all, blocking
        socket.writeFully(requestBuffer);
        requestBuffer.clear();
    }
```
Also: `writeRequestInPipeline` isn't used in any tests. Although it wasn't before either.